In [10]:
import pandas as pd
 
 
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)

In [11]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


Pregunta A: 

75% = 31.000000 

Max = 512.329200

Diferencia de 41.3292 

Esto nos dice que hay una depresion en los precios, algunos pajaron tarifas más altas que otros.


Pregunta B:

Count PassengerId: 891.000000

Count Age: 714.000000

Diferencia de 177

Esto nos dice que 177 pasajeros no tiene un valor registrado en Age, lo que nos dice que la anamalia es que existen datos faltanes en la columna Age.


Pregunta C:

- Reemplazar los valores faltantes de Age pr la mediana de las edades
- Eliminar las filas de pasajeros que no tengan Age, aunque esto la cantidad de datos disponibles

In [ ]:
# 26 - Ago
def female_fare():
    female = df[df['Sex'] == 'female']
    return female['Fare'].mean()

print(f"El promedio de las mujeres es: {female_fare()}")

def male_fare():
    male = df[df['Sex'] == 'male']
    return male['Fare'].mean()

print(f"El promedio de los hombres es: {male_fare()}")

El promedio de las mujeres es: 44.47981783439491
El promedio de los hombres es: 25.523893414211443


In [24]:
df.groupby('Sex')['Fare'].mean()

Sex
female    44.479818
male      25.523893
Name: Fare, dtype: float64

In [34]:
df['Fare'].value_counts

<bound method IndexOpsMixin.value_counts of 0       7.2500
1      71.2833
2       7.9250
3      53.1000
4       8.0500
        ...   
886    13.0000
887    30.0000
888    23.4500
889    30.0000
890     7.7500
Name: Fare, Length: 891, dtype: float64>

In [38]:
primer_cuartil = df['Fare'].quantile(0.25)
tercer_cuartil = df['Fare'].quantile(0.75)

IQR = tercer_cuartil - primer_cuartil
minimun = primer_cuartil - 1.5*IQR
maximum = tercer_cuartil + 1.5*IQR


print("Primer cuartil: ", primer_cuartil)
print("Tercer cuartil: ", tercer_cuartil)
print("IQR: ", IQR)
print("Mínimo: ", minimun)
print("Máximo: ", maximum)

Primer cuartil:  7.9104
Tercer cuartil:  31.0
IQR:  23.0896
Mínimo:  -26.724
Máximo:  65.6344
